### PR TITLE
Minor FiftyOne integration tweaks

### PR DESCRIPTION
**Pull request by ehofesmann**
## What does this PR do?

This PR adds a couple small tweaks to the FiftyOne integration, primarily to the documentation:
- Documenting the various environments in which the `visualize()` method can be used (shell, script, Jupyter notebook, Colab, remote, etc)
- Adding a section to the README on visualizing outputs of vision tasks
- Updating the embeddings example and image to `resnet101` to match other flash embeddings examples. @ethanwharris Please replace the [previous embeddings image](https://pl-flash-data.s3.amazonaws.com/assets/fiftyone/embeddings.png) you uploaded with this one if possible: https://user-images.githubusercontent.com/21222883/122239476-f79ef600-ce8e-11eb-814c-910fbc6d07ed.png

Here's the content of the new README section (all Flash code)

## Visualization

Predictions from image and video tasks can be visualized through an [integration with FiftyOne](https://lightning-flash.readthedocs.io/en/latest/integrations/fiftyone.html), allowing you to better understand and analyze how your model is performing.

```python
from flash.core.data.utils import download_data
from flash.core.integrations.fiftyone import visualize
from flash.image import ObjectDetector
from flash.image.detection.serialization import FiftyOneDetectionLabels

# 1. Download the data
# Dataset Credit: https://www.kaggle.com/ultralytics/coco128
download_data(
    "https://github.com/zhiqwang/yolov5-rt-stack/releases/download/v0.3.0/coco128.zip",
    "data/",
)

# 2. Load the model from a checkpoint and use the FiftyOne serializer
model = ObjectDetector.load_from_checkpoint(
    "https://flash-weights.s3.amazonaws.com/object_detection_model.pt"
)
model.serializer = FiftyOneDetectionLabels()

# 3. Detect the object on the images
filepaths = [
    "data/coco128/images/train2017/000000000025.jpg",
    "data/coco128/images/train2017/000000000520.jpg",
    "data/coco128/images/train2017/000000000532.jpg",
]
predictions = model.predict(filepaths)

# 4. Visualize predictions in FiftyOne App
session = visualize(predictions, filepaths=filepaths)
```

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/core/integrations/fiftyone/utils.py [hunk 1]
```diff
@@ -20,51 +20,72 @@
...
 
     Args:
-        labels: Either a list of FiftyOne labels that will be applied to the
-            corresponding filepaths provided with through `filepath` or
-            `datamodule`. Or a list of dictionaries containing image/video
-            filepaths and corresponding FiftyOne labels.
+        predictions: Can be either a list of FiftyOne labels that will be
+            matched with the corresponding ``filepaths``, or a list of
+            dictionaries with "filepath" and "predictions" keys that contains
+            the filepaths and predictions. The
```
Comment (ethanwharris) [inline flash/core/integrations/fiftyone/utils.py:53]:
```suggestion
            the filepaths and predictions.
```

File: flash/core/integrations/fiftyone/utils.py [hunk 2]
```diff
@@ -20,51 +20,72 @@
...
         filepaths: A list of filepaths to images or videos corresponding to the
-            provided `labels`.
-        wait: A boolean determining whether to launch the FiftyOne session and
-            wait until the session is closed or whether to return immediately.
-        label_field: The string of the label field in the FiftyOne dataset
-            containing predictions
+            provided ``predictions``
```
Comment (ethanwharris) [inline flash/core/integrations/fiftyone/utils.py:55]:
```suggestion
            provided ``predictions``.
```

File: flash/core/integrations/fiftyone/utils.py [hunk 3]
```diff
@@ -20,51 +20,72 @@
...
+        label_field: The name of the label field in which to store the
+            predictions in the FiftyOne dataset
+        wait: Whether to block execution until the FiftyOne App is closed
+        **kwargs: Optional keyword arguments for
+            :meth:`fiftyone:fiftyone.core.session.launch_app`
```
Comment (ethanwharris) [inline flash/core/integrations/fiftyone/utils.py:60]:
```suggestion
            :meth:`fiftyone:fiftyone.core.session.launch_app`.
```

File: flash/core/integrations/fiftyone/utils.py [hunk 4]
```diff
@@ -20,51 +20,72 @@
...
         filepaths: A list of filepaths to images or videos corresponding to the
-            provided `labels`.
-        wait: A boolean determining whether to launch the FiftyOne session and
-            wait until the session is closed or whether to return immediately.
-        label_field: The string of the label field in the FiftyOne dataset
-            containing predictions
+            provided ``predictions``
+        label_field: The name of the label field in which to store the
+            predictions in the FiftyOne dataset
+        wait: Whether to block execution until the FiftyOne App is closed
```
Comment (ethanwharris) [inline flash/core/integrations/fiftyone/utils.py:58]:
```suggestion
        wait: Whether to block execution until the FiftyOne App is closed.
```

File: flash/core/integrations/fiftyone/utils.py [hunk 5]
```diff
@@ -20,51 +20,72 @@
...
+            the filepaths and predictions. The
         filepaths: A list of filepaths to images or videos corresponding to the
-            provided `labels`.
-        wait: A boolean determining whether to launch the FiftyOne session and
-            wait until the session is closed or whether to return immediately.
-        label_field: The string of the label field in the FiftyOne dataset
-            containing predictions
+            provided ``predictions``
+        label_field: The name of the label field in which to store the
+            predictions in the FiftyOne dataset
```
Comment (ethanwharris) [inline flash/core/integrations/fiftyone/utils.py:57]:
```suggestion
            predictions in the FiftyOne dataset.
```

File: flash/core/integrations/fiftyone/utils.py [hunk 6]
```diff
@@ -20,51 +20,72 @@
...
+        default web browser
+    -   **Remote Python shell**: Pass the ``remote=True`` option to this method
+        and then follow the instructions printed to your remote shell to open
+        the App in your browser on your local machine
+    -   **Jupyter notebook**: The App will launch in the output of your current
+        cell
+    -   **Google Colab**: The App will launch in the output of your current
+        cell
+    -   **Python script**: Pass the ``wait=True`` option to block execution of
+        your script until the App is closed
```
Comment (ethanwharris) [inline flash/core/integrations/fiftyone/utils.py:44]:
```suggestion
    -   **Local Python shell**: The App will launch in a new tab in your
        default web browser.
    -   **Remote Python shell**: Pass the ``remote=True`` option to this method
        and then follow the instructions printed to your remote shell to open
        the App in your browser on your local machine.
    -   **Jupyter notebook**: The App will launch in the output of your current
        cell.
    -   **Google Colab**: The App will launch in the output of your current
        cell.
    -   **Python script**: Pass the ``wait=True`` option to block execution of
        your script until the App is closed.
```

File: flash/image/segmentation/serialization.py [hunk 7]
```diff
@@ -89,22 +90,23 @@ def serialize(self, sample: Dict[str, torch.Tensor]) -> torch.Tensor:
...
 class FiftyOneSegmentationLabels(SegmentationLabels):
+    """A :class:`.Serializer` which converts the model outputs to FiftyOne
+    segmentation format.
+
+    Args:
+        labels_map: A dictionary that map the labels ids to pixel intensities.
+        visualize: whether to visualize the image labels.
+        return_filepath: Boolean determining whether to return a dict
+            containing filepath and FiftyOne labels (True) or only a list of
+            FiftyOne labels (False)
```
Comment (ethanwharris) [inline flash/image/segmentation/serialization.py:101]:
```suggestion
            FiftyOne labels (False).
```

**Review by ethanwharris — APPROVED** [review]
@ehofesmann Nice! I've updated the embeddings image. Other than that just minor comments

**Comment by ehofesmann** [comment]
> @ehofesmann Nice! I've updated the embeddings image. Other than that just minor comments

Awesome, thanks a lot!

**Comment by ethanwharris** [comment]
@ehofesmann We've been getting this error quite a bit today on docs builds:
```
intersphinx inventory 'https://voxel51.com/docs/fiftyone/objects.inv' not fetchable due to <class 'requests.exceptions.ConnectionError'>: HTTPSConnectionPool(host='voxel51.com', port=443): Max retries exceeded with url: /docs/fiftyone/objects.inv (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7eff1a6bb1d0>: Failed to establish a new connection: [Errno 110] Connection timed out'))
```
It usually is fine if I just re-trigger the build, but that's not really sustainable. Do you have any idea what might be causing the issue?

**Comment by ehofesmann** [comment]
> @ehofesmann We've been getting this error quite a bit today on docs builds:
> 
> ```
> intersphinx inventory 'https://voxel51.com/docs/fiftyone/objects.inv' not fetchable due to <class 'requests.exceptions.ConnectionError'>: HTTPSConnectionPool(host='voxel51.com', port=443): Max retries exceeded with url: /docs/fiftyone/objects.inv (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7eff1a6bb1d0>: Failed to establish a new connection: [Errno 110] Connection timed out'))
> ```
> 
> It usually is fine if I just re-trigger the build, but that's not really sustainable. Do you have any idea what might be causing the issue?

We've spent some time looking into this and haven't been able to reproduce those issues locally or in our own fork's actions. We did push website updates yesterday and it is possible that our hosting provider may have a hiccup today. Let me know if this keeps showing up and we will continue to investigate it.

**Comment by ethanwharris** [comment]
@ehofesmann cool, will do, thanks for investigating :smiley:

**Pull request merged**